src/CameraOnvif.hpp

```diff
@@ -0,0 +1,48 @@
+#ifndef CAMERA_ONVIF_CAMERAONVIF_HPP
+#define CAMERA_ONVIF_CAMERAONVIF_HPP
+
+#include <string>
+
+namespace camera_onvif {
+    /**
+     *
+     */
+    class CameraOnvif {
+        struct Private;
+        Private* m_private = nullptr;
```

Comment:
What's Private and m_private? Please, always try to give meaningful names to members, this makes your code a lot easier to read by others...

Reply:
In this case, I'm with him. The goal of Private is to avoid leaking the whole SOAP generated code
to the user of the library (i.e. Private is a good a name as any).

Would need some commenting though for sure. Would just adding a comment be fine ?

Reply:
I guess so, yes

Reply:
So, is commenting "The goal of Private is to avoid leaking the whole SOAP generated code" enough for you?

Reply:
Yes, it is. But, this is NOT what the code is doing... You have to use `CameraOnvif::Private`'s destructor to free the resources you have allocated

Reply:
I added the destructor yesterday.